tools/lint-docstring

```diff
@@ -0,0 +1,183 @@
+
+
+def create_file_overview_doc() -> None:
+    """
+    Creates the entire developer-file-overview.md document again
+    by retaining the lines 1-6 and the rest from the docstrings in the files
+    """
+    folder_file_docstring = extract_doctrings_from_files()
+    with open(DEVELOPER_DOC, "r+") as file:
+        doc_data = file.readlines()
+
+    # writing operation starts at index 7 as the table content starts from line 7
+    write_location = INITIAL_INDEX
+    doc_to_write = doc_data[: INITIAL_INDEX - 1]
+    files_and_docstrings_to_add = []
+
+    for folder in folder_file_docstring:
+        folder_name_added = False
+        dictionary_of_files = folder_file_docstring[folder]
+
+        folder_text = folder
+        for file in sorted(dictionary_of_files):
+            # description_text = folder_file_docstring[folder][file]
+            docstring = folder_file_docstring[folder][file]
+            if folder_name_added is True:
+                # only the first row of a folder has the folder name, rest contain whitespace
+                folder_text = " "
+            file_text = file
+            text_tuple = (folder_text, file_text, docstring)
+            folder_name, file_name, desc = column_text_with_spaces(
+                text_tuple, COLUMN_WIDTHS
+            )
+            new_line = f"| {folder_name}| {file_name}| {desc}|\n"
+            files_and_docstrings_to_add.append(new_line)
+            folder_name_added = True
+
+        # adding blank row at the end of every folder
+        files_and_docstrings_to_add.append(BLANK_LINE)
+
+    # Folders that do not contain any files with docstrings are added separately to the file-overview
+    for skipped_folder_name in sorted(DOCSTRINGS_FOR_NO_FILE_FOLDERS):
+        skipped_folder_tuple = (
+            skipped_folder_name,
+            " ",
+            DOCSTRINGS_FOR_NO_FILE_FOLDERS[skipped_folder_name],
+        )
+        folder_name, file_name, desc = column_text_with_spaces(
+            skipped_folder_tuple, COLUMN_WIDTHS
+        )
+        new_line = f"| {folder_name}| {file_name}| {desc}|\n"
+        files_and_docstrings_to_add.append(new_line)
+        files_and_docstrings_to_add.append(BLANK_LINE)
+    write_to_file(files_and_docstrings_to_add[:-1], doc_to_write, write_location)
+
+
+def extract_doctrings_from_files() -> Dict[str, Dict[str, str]]:
+    """
+    Returns a dictionary containing folder name which in turn
+    is a dictionary containing files and their respective descriptons
+    """
+    total_files = extract_folder_file_structure()
+    folder_file_docstring: Dict[str, Dict[str, str]] = {}
+    for folder, files in sorted(total_files.items()):
+        folder_file_docstring[str(folder)] = {}
+        for file in files:
+            imported_file = importlib.import_module(
+                f'{folder.replace("/",".")}.{file[:-3]}'
+            )
+            extracted_docstring = cast(str, imported_file.__doc__)
```

Comment:
Before you cast, check what type it is, and whether you really need to cast. Then if you do, I'd recommend always adding a comment unless it's really obvious.

What is the type here? Why did you add a cast, given we're not running mypy on this by default?

Reply:
Okay.

Actually, I tried being overly cautious regarding the type and hence kept the cast function. I tested the code without the cast and it works well. I'll remove it in the updated commit(s).

Reply:
By using `reveal_type` with `mypy`, this is apparently `Optional[str]` - this may be important since some modules may end up having no docstring :)